kasa/deviceconfig.py

```diff
@@ -214,22 +175,30 @@ def __post_init__(self) -> None:
                 DeviceFamily.IotSmartPlugSwitch, DeviceEncryptionType.Xor
             )
 
-    def to_dict(
+    def __pre_serialize__(self) -> Self:
+        return replace(self, http_client=None)
```

Comment:
Back to my comment above, I somehow doubt that there's a nicer way to mark fields as "non-exported". Any ideas?

Reply:
As above, none I'm afraid but will raise a ticket.